### PR TITLE
[FSK-13] : Configure react-location

### PR DESCRIPTION
**Pull request by erfan-goodarzi**
This PR add React location to project.

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,5 +1,13 @@
+import { ReactLocation, Router } from '@tanstack/react-location';
+
```
Comment (AmirabbasJ) [inline src/App.tsx:2]:
```suggestion

const location = new ReactLocation();
```

I think it's a better idea we put this outside of app 
what you think?

File: src/App.tsx [hunk 2]
```diff
@@ -1,5 +1,13 @@
...
 const App: React.FC = () => {
-  return <div>The Charity App</div>;
+  const location = new ReactLocation();
+  return (
+    <>
+      The Charity App
+      <Router routes={[{ path: '/' }]} location={location} />
+    </>
+  );
```
Comment (AmirabbasJ) [inline src/App.tsx:10]:
```suggestion
  return (
     <Router routes={[{ path: '/' }]} location={location}>
        The Charity App
    </Router>
  );
```

why not do this instead?
we get rid of the wrapping `<> ... </>` plus I've seen them doing it this way 
what are your thoughts?

Comment (AmirabbasJ) [inline src/App.tsx:10]:
```suggestion
  return (
     <Router routes={[{ path: '/' }]} location={location}>
        The Charity App
    </Router>
  );
```

why not do this instead?
we get rid of the wrapping `<> ... </>` plus I've seen them doing it this way 
what are your thoughts?

File: src/App.tsx (hunk 1 above)
Comment (AmirabbasJ) [inline src/App.tsx:2]:
```suggestion

const location = new ReactLocation();
```

I think it's a better idea if we put this outside of app 
what you think?

**Review by AmirabbasJ — CHANGES_REQUESTED** [review]

File: src/App.tsx (hunk 2 above)
Comment (AmirabbasJ) [reply src/App.tsx:10]: like what exactly?
I don't think we have anything as a sibling in the app, maybe just a wrapper

**Comment by AmirabbasJ** [comment]
And one thing to note that this merge should happen after #2 
so I'd say even after getting approved, you wait till that pull request is merged, then pull the updates to this branch
and then merge it

**Comment by ASafaeirad** [comment]
#2 is merged

**Pull request merged**